litestar/datastructures/headers.py

```diff
@@ -352,14 +384,70 @@ def prevent_storing(cls) -> "CacheControlHeader":
 
         return cls(no_store=True)
 
+    @classmethod
+    def _get_type_annotations(cls) -> Dict[str, Type]:
```

Comment:
so, i think you are reinventing the wheel here. Please rebase on main and than use the logic inside `litestar.typing` to handle this kind of stuff. There is a class there called `FieldDefinition` and it is used throughout the codebase to parse types.